### PR TITLE
Move loading indicator to the top, so it's actually visible when the competitions list is long

### PR DESCRIPTION
**Pull request by jfly**
Before:

image

After:

image

@jonatanklosko, could you take a quick look? Thanks!

### CODE REVIEW

**Comment by jonatanklosko** [comment]
Actually that's what I've done before but then realised that it would be better to have the spinner in the middle of the map, otherwise it looks kinda bad. I'm not sure which option would be better.

**Comment by jfly** [comment]
Ah, darn. I see what you mean. I tweaked the code a bit so we can style the loading indicator differently based on if we're showing the map or the list. What do you think?

File: WcaOnRails/app/views/competitions/index.html.erb [hunk 1]
```diff
@@ -20,7 +20,3 @@
     </div>
   </div>
 </div>
-
-<script>
-  $('#competitions-<%= params[:display] == "list" ? "map" : "list" %>').hide();
-</script>
```
Comment (jonatanklosko) [inline WcaOnRails/app/views/competitions/index.html.erb]: Why did you remove that? It shows the list/map container accordingly.

File: WcaOnRails/app/views/competitions/index.js.erb [hunk 2]
```diff
@@ -2,10 +2,5 @@ history.pushState(null, document.title, <%= current_url.to_json.html_safe %>);
 
 $('#competitions-list').html('<%=j render 'index_competitions_list', competitions: @competitions %>');
 
-<% if params[:display] == "list" %>
-  $('#competitions-map').hide();
-  $('#competitions-list').show();
-<% else %>
-  $('#competitions-list').hide();
-  $('#competitions-map').show();
-<% end %>
```
Comment (jonatanklosko) [inline WcaOnRails/app/views/competitions/index.js.erb]: Same as above.

File: WcaOnRails/app/views/competitions/index.html.erb (hunk 1 above)
Comment (jonatanklosko) [inline WcaOnRails/app/views/competitions/index.html.erb]: One bug that I see here. You removed this piece of code but didn't add the appropriate class after an html request.

Comment (jfly) [inline WcaOnRails/app/views/competitions/index.html.erb]: Really? That's what I'm trying to do over in WcaOnRails/app/views/competitions/index.js.erb. Did I miss something?

Comment (jonatanklosko) [inline WcaOnRails/app/views/competitions/index.html.erb]: The index.js.erb is loaded only after Ajax request, but the first time you load the page html request is sent so the script wouldn't be executed.

Comment (jfly) [inline WcaOnRails/app/views/competitions/index.html.erb]: Ah, what about line 11 of this file?

```html
+  <div id="search-results" class="row competitions-<%= params[:display] == "list" ? "list" : "map" %>">
```

Comment (jonatanklosko) [inline WcaOnRails/app/views/competitions/index.html.erb]: Oh, my bad you achieved it via css classes =)

**Comment by jonatanklosko** [comment]
Awesome cleanup with these css classes, LGTM!

**Comment by jfly** [comment]
Thanks, @jonatanklosko!
